server.js

```diff
@@ -4,20 +4,104 @@ const cors = require("cors");
 const app = express();
 
 app.use(cors());
+app.use(express.json());
 
 const welcomeMessage = {
   id: 0,
   from: "Bart",
   text: "Welcome to CYF chat system!",
 };
 
-//This array is our "data store".
-//We will start with one message in the array.
-//Note: messages will be lost when Glitch restarts our server.
-const messages = [welcomeMessage];
+// This array is our "data store".
+// We will start with one message in the array.
+// Note: messages will be lost when Glitch restarts our server.
+const messages = [
+  welcomeMessage,
+  {
+    id: 1,
+    from: "BAki",
+    text: "CAn you see me?",
+  },
+];
 
 app.get("/", function (request, response) {
   response.sendFile(__dirname + "/index.html");
 });
 
-app.listen(process.env.PORT);
+app.get("/messages", (request, response) => {
+  response.send({ messages });
+});
+
+// Send a message
+app.post("/messages", (request, response) => {
+  let newMessage = request.body;
+  if (
+    newMessage.id === "" ||
+    newMessage.from === "" ||
+    newMessage.text === ""
+  ) {
+    console.error("Invalid message:", newMessage);
+    response.status(400).send("Bad Request");
+  } else {
+    messages.push(newMessage);
+    response.send("Message posted successfully");
+  }
+});
+
+// Get messages containing the search word
+app.get("/messages/search", (request, response) => {
+  let searchWord = request.query.text;
+  response.send(getMessagesFromSearch(messages, searchWord));
+});
+
+// Get the 10 latest messages
+app.get("/messages/latest", (request, response) => {
+  response.send(getLatestMessages(messages));
+});
+
+// Get one message by ID
+app.get("/messages/:id", (request, response) => {
+  let messageId = request.params.id;
+  response.send(getMessageByID(messages, messageId));
+});
+
+// Delete message by ID
+app.delete("/messages/:id", (request, response) => {
+  let messageId = request.params.id;
+  response.send(deleteMessageByID(messages, messageId));
+});
+
+app.listen(process.env.PORT, () => {
+  console.log(`Listening on PORT ${process.env.PORT}...`);
+});
+
+// Helper function to get a message by ID
+const getMessageByID = (messages, id) => {
+  return messages.filter((message) => message.id == id);
```

Comment:
FYI. Another way to store the message is to use Map(), so you can easily get item by id.
But it is not for search.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map